client/src/main/java/org/threadly/litesockets/client/http/HTTPClient.java

```diff
@@ -520,12 +518,14 @@ public void onRead(Client client) {
     private TCPClient client;
     private long lastRead = Clock.lastKnownForwardProgressingMillis();
 
-    public HTTPRequestWrapper(ClientHTTPRequest chr) {
+    private HTTPRequestWrapper(ClientHTTPRequest chr) {
       hrp = new HTTPResponseProcessor(chr.getHTTPRequest()
                                          .getHTTPRequestHeader()
                                          .getRequestMethod().equals("HEAD"));
       hrp.addHTTPResponseCallback(this);
       this.chr = chr;
+      
+      sei.watchFuture(slf, chr.getTimeoutMS());
```

Comment:
Might need to change the documentation on this, pretty sure I wrote somewhere the request timeout only starts when the actual request is sent to the server (or on the socket anyway).  I didnt think it could go negative, though it could wait for a long time if the queue was large, though I might have messed that up sometime later.

Reply:
The situation I witnessed was because the queue was not making progress because the remote service had hung.  So there was definitely a service issue, the problem was it was compounded by clients not giving up on it.

> pretty sure I wrote somewhere the request timeout only starts when the actual request is sent to the server

I did another look around and not finding any docs like this.  So I assume you are good for me to go ahead and merge / release this.